storage/innobase/btr/btr0pcur.cc

```diff
@@ -420,12 +420,27 @@ btr_pcur_move_to_next_page(
 	cursor->old_stored = BTR_PCUR_OLD_NOT_STORED;
 
 	page = btr_pcur_get_page(cursor);
+	const ulint page_no = buf_block_get_page_no(buf_block_align(page));
+
 	next_page_no = btr_page_get_next(page, mtr);
 	space = buf_block_get_space(btr_pcur_get_block(cursor));
 	zip_size = buf_block_get_zip_size(btr_pcur_get_block(cursor));
 
 	ut_ad(next_page_no != FIL_NULL);
 
+	if (next_page_no - page_no == 1) {
```

Comment:
Factor out this to a function with two page no and a page args, call it from btr_pcur_move_backward_from_page too

Reply:
Common code extracted into `btr_update_scan_stats()`